src/main.rs

```diff
@@ -585,80 +586,100 @@ impl ManParser for Type1 {
+                let option_desc = data[1].trim().replace('\n', " ");
+                built_command(
+                    option_name,
+                    option_desc.as_str(),
+                    built_command_output,
+                    existing_options,
+                    cmd_name,
+                );
+            } else {
+                // add_diagnostic(format!("{:?} does not contain '-'", option_name));
+                return false;
+            }
+            // XXX possible to add fallback2 here
+
+            options_section = &options_section[mat.get(0).unwrap().end() - 3..];
+            options_matched = options_parts_re.captures(options_section);
+        }
+        true
     }
-}
 
-//     def fallback2(self, options_section):
-//         add_diagnostic('Trying last chance fallback')
-//         ix_remover_regex = re.compile("\.IX.*")
-//         trailing_num_regex = re.compile('\\d+$')
-//         options_parts_regex = re.compile("\.IP (.*?)\.IP", re.DOTALL)
-//
-//         options_section = re.sub(ix_remover_regex, "", options_section)
-//         options_matched = re.search(options_parts_regex, options_section)
-//         if options_matched == None:
-//             add_diagnostic('Still (still!) not found')
-//             return False
-//         while options_matched != None:
-//             data = options_matched.group(1)
-//
-//             data = remove_groff_formatting(data)
-//             data = data.strip()
-//             data = data.split("\n",1)
-//             if (len(data)>1 and len(data[1].strip())>0): # and len(data[1])<400):
-//                 optionName = re.sub(trailing_num_regex, "", data[0].strip())
-//
-//                 if ('-' not in optionName):
-//                     add_diagnostic("%r doesn't contain '-'" % optionName)
-//                 else:
-//                     optionName = optionName.strip()
-//                     optionName = unquote_double_quotes(optionName)
-//                     optionName = unquote_single_quotes(optionName)
-//                     optionDescription = data[1].strip().replace("\n"," ")
-//                     built_command(optionName, optionDescription)
-//             else:
-//                 add_diagnostic('Unable to split option from description')
-//                 return False
-//
-//             options_section = options_section[options_matched.end()-3:]
-//             options_matched = re.search(options_parts_regex, options_section)
-//         return True
+    fn fallback2(
+        &self,
+        options_section: &str,
+        built_command_output: &mut Vec<String>,
+        existing_options: &mut HashSet<String>,
+        cmd_name: &str,
+    ) -> bool {
+        // add_diagnostic("Trying last chance fallback");
+        let ix_remover_re = Regex::new(r"\.IX.*").unwrap();
+        let trailing_num_re = Regex::new(r"\d+$").unwrap();
+        let options_parts_re = Regex::new(r"\.IP ((?s:.)*?)\.TP").unwrap();
```

Comment:
typo?
```suggestion
        let options_parts_re = Regex::new(r"\.IP ((?s:.)*?)\.IP").unwrap();
```

Reply:
Yup, nice findings. I probably just copied the one above.